### ISSUE TITLE
VS Extension attempts to download package via HTTP when source is HTTPS

### ISSUE THREAD

**Issue by rossipedia** [NuGet/Home#1080]
At StackExchange we have an internal nuget feed supplied by our TeamCity server, which is hosted on an HTTPS endpoint.

Using the VS Extension I can browse the packages supplied by the feed just fine, but run into issues when trying to install / update them.

Here's the error message I get when trying to upgrade a package: 

```
NuGet.Protocol.Core.Types.NuGetProtocolException: An error occurred while downloading package 'StackExchange.Network 1.0.49' from source 'https://build.stackexchange.com/httpAuth/app/nuget/v1/FeedService.svc/'. ---> System.Net.WebException: The remote server returned an error: (401) Unauthorized.
```
The output from `nuget sources`: 

```
> nuget sources list
Registered Sources:

  1.  TeamCity [Enabled]
      https://build.stackexchange.com/httpAuth/app/nuget/v1/FeedService.svc/
  2.  nuget.org v2 [Enabled]
      https://www.nuget.org/api/v2/
```

I also have my credentials for the TeamCity encrypted in my `Nuget.config` file, which I believe is how searching and browsing packages works just fine.

Here's the Fiddler capture from the above attempted upgrade:

![image](https://cloud.githubusercontent.com/assets/96213/8989102/3a14db66-36a7-11e5-8e9e-b84ba1a6936e.png)

Since my source url is `https`, shouldn't the request for the actual package be an `https` request and not `http`?

It seems to be exhibiting the same thing as #1079 which causes the download of the package to fail, since the redirect from HTTP -> HTTPS somehow loses my credentials.

**Comment by yishaigalatzer** [comment]
Looks like you are using an extremely old version of NuGet (v1), not to say this is related to the issue you are seeing, but this is a protocol we would love to deprecate in the next few years. Its probably a good idea to upgrade your server or contact the authors of the server.

Regardless we will look at the issues you are reporting here and get back to you

**Comment by yishaigalatzer** [comment]
To help us figure out if this is a NuGet issue or a server issue, can you please look into the payload that defines the download source? Should be one of the previous entries on the fiddler trace. 

What I'm suspecting this to be, is that the server might be returning an http url for download, and nuget.exe just follows the link. If this is the case the culprit is the server.

**Comment by rossipedia** [comment]
> What I'm suspecting this to be, is that the server might be returning an http url for download, and nuget.exe/visual studio extension just follows the link. If this is the case the culprit is the server.

I looked at another Fiddler trace, and this is indeed the case. However, shouldn't the original `Authorization` header be supplied to the new URL in response to a `302` response?

**Comment by rossipedia** [comment]
I also just verified that our nuget feed is indeed using v2 of the api (even though the URL contains `/v1`, it's actually v2)

**Comment by pranavkm** [comment]
What happens when nuget attempts to connect to the HTTP url? Does it 401 pointing to an HTTPS endpoint?

**Comment by rossipedia** [comment]
Yeah, sorry, I think this was resolved in #1079. The `Authorization` header handling still seems a bit suspect, but I'm currently functional.
